.changeset/checkbox-remove-legacy.md

```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/checkbox': major
+---
+
+Removes `LegacyCheck`, previously deprecated an inaccessible since v7.0
```

Comment:
`deprecated and inaccessible`?

Reply:
Not only was it deprecated, but it was difficult to even import (maybe "inaccessible is the wrong word")

Reply:
Yeah, I was pointing out the `and`. Next time, I'll add a suggestion for stuff like this.